internal/controller/kuadrant_controller_test.go

```
@@ -0,0 +1,85 @@
+
+	// We can't directly create a controller.Controller, so we'll just test the logic
+	// In a real scenario, the KuadrantController wraps a real controller
+
+	// Simulate the logic from KuadrantController.Reconcile
+	result, err := mock.Reconcile(context.Background(), ctrlruntimereconcile.Request{})
+
+	// Blocking error should be returned immediately
+	if err == nil {
+		t.Error("Expected blocking error to be returned")
+	}
+	if err != blockingErr {
```

golangci-lint (2.12.2)

[error] comparing with != will fail on wrapped errors. Use errors.Is to check for a specific error

(errorlint)